torch/csrc/jit/tracer.h

```diff
@@ -62,8 +62,38 @@ inline std::vector<VariableFlags> getVarFlags(const variable_list& vars) {
   return fmap(vars, &VariableFlags::of);
 }
 
-}
+} // namespace detail
+
+
+struct ArgumentStash {
```

Comment:
Want a quick explanation here what an ArgumentStash is and what it's used for

Reply:
So the issue with tracing of e.g. sizes is that by the time they get to where we record the trace, they are already `IntList`, so we can't associate any meta data with them. To get around this, the argument parser is augmented to stash any arguments containing traced values in here, and the code that emits the IR node is adjusted to add dependencies on them if the stash is not empty.

I'll add this as a comment too.